### DISCUSSION TITLE
What kind of dynamic economy we are going for?

### DISCUSSION THREAD

**Discussion by 12gabriel3** [EveronLife/EveronLife#149] [General]
# Dynamic economies

A dynamic economy is a pretty general concept, it means that the prices will be reactive to player activity, simulating or implementing supply and demand. I've come up with a few _base_ ways that it can be done. These could be mixed together for different results

## Supply reactive prices

Supply reactive prices takes only supply into account and simulates demand. This is the most simple form of player driven economy that I can think of. The prices slowly ticks up, simulating demand, and the more items are sold by players, the more the price goes down.

## Supply and demand with infinite liquidity

With this approach, the only simulated part is liquidity. This means that players create both supply _and_ demand for the item, but _liquidity_ is infinite, meaning that items cannot be _out of stock_ and you can sell the items immediately. With this approach, every item must have _use value_ for the players, to create the demand. This means consumable drugs, maintenance on cars, fuel consumption. In short, every item must have a continuous use, either by consuming, by losing on death, by requiring maintenance or by using it to produce another item.

## Fully player driven economy

This is the same as the latter approach, but the _liquidity_ is also created by players. The economy is comprised by both buy and sell orders, like [Eve](https://en.wikipedia.org/wiki/Eve_Online) or [Albion](https://en.wikipedia.org/wiki/Albion_Online). This means you can only sell something immediately if another player is willing to buy it and vice versa. Otherwise you need to create a buy or sell order. This also creates a new way of providing liquidity: through trading, which is very interesting.

## My thoughts

I really like the idea of a fully player driven economy, with buy and sell orders. Even though it is used mainly by MMOs, I think we could have similar results by having less items. We could achieve that by making several player used items consume from the same pool of market items. This should solve liquidity issues that arise from a small player base. Example:

* M16: 100 weapon parts
* M9: 20 weapon parts
* RPG: 500 weapon parts

Basically every weapon could consume from the same weapon parts pool. Same for other items. Clothing items could consume from a textiles pool, for example. 

Leave a comment on what are your thoughts for the game economy

**Comment by Arkensor** [comment]
A Player driven economy is difficult as we have no idea yet how many players will usually play on one server at any given time and how diverse the resources they farm are. Dynamic market prices are certainly something we want to end up looking into but its unlikely we will add anything like it this year. It is worth starting to think about it though.

My idea for it was to have some supply and demand pricing that has a limited min and max. You would configure what prices are influenced when you buy or sell an item, and I think that would make sense to use "shared pools" as you propose, so any weapon transaction increases or decreases an amount per weapon configured and then each item can define factors for all pools it depends on to determine the overall price. e.g. An AK always costs $5000 min, but depends on metal, wood and something else so if all those are hyper-expensive it might cost up to $10000.

# Player to player trading
That is about dynamic markets though. An "optional player-driven economy" is something else we can achieve by encouraging RP interaction to buy and sell items in RP and not always just through nps. e.g.:
Player A chops wood, processes it into lumber and drives to a trader to sell it. Player B needs lumber for house construction and he also drives to the trader, but he wants to buy instead. Player A could sell one piece of lumber for $100 to the NPC, and player B could buy lumber there for $200. If the two communicate and sort out a price on their own Player A could get more money for parts or all of his farming run by selling to player B for $150 each and player B saved $50 per lumber he wanted to buy. So both have a financial benefit and they had the chance for social interaction - which is after all the end goal. The prices are of course ab it extreme, but I wanted to explain the concept. Together with dynamic pricing and smaller ranges between sell and buy prices on npcs it should work.

**Comment by jasper-one** [comment]
Added some of my thoughts / what I think we will require first to make this a reality here #167 @12gabriel3
